synapseclient/models/submission.py

```diff
@@ -0,0 +1,939 @@
+
+        Returns:
+            The Submission object.
+        """
+        self.id = synapse_submission.get("id", None)
+        self.user_id = synapse_submission.get("userId", None)
+        self.submitter_alias = synapse_submission.get("submitterAlias", None)
+        self.entity_id = synapse_submission.get("entityId", None)
+        self.version_number = synapse_submission.get("versionNumber", None)
+        self.evaluation_id = synapse_submission.get("evaluationId", None)
+        self.evaluation_round_id = synapse_submission.get("evaluationRoundId", None)
+        self.name = synapse_submission.get("name", None)
+        self.created_on = synapse_submission.get("createdOn", None)
+        self.team_id = synapse_submission.get("teamId", None)
+        self.contributors = synapse_submission.get("contributors", [])
+        self.submission_status = synapse_submission.get("submissionStatus", None)
+        self.entity_bundle_json = synapse_submission.get("entityBundleJSON", None)
+        self.docker_repository_name = synapse_submission.get(
+            "dockerRepositoryName", None
+        )
+        self.docker_digest = synapse_submission.get("dockerDigest", None)
+
+        return self
+
+    async def _fetch_latest_entity(
+        self, *, synapse_client: Optional[Synapse] = None
+    ) -> dict:
+        """
+        Fetch the latest entity information from Synapse.
+
+        <https://rest-docs.synapse.org/rest/GET/entity/id.html>
+
+        If the object is a DockerRepository, this will also fetch the DockerCommit object with the latest createdOn value
+        and attach it to the final dictionary:
+
+        <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/docker/DockerCommit.html>
+
+        Arguments:
+            synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+        Returns:
+            Dictionary containing entity information from the REST API.
+
+        Raises:
+            ValueError: If entity_id is not set or if unable to fetch entity information.
+        """
+        if not self.entity_id:
+            raise ValueError("entity_id must be set to fetch entity information")
+
+        from synapseclient import Synapse
+
+        client = Synapse.get_client(synapse_client=synapse_client)
+
+        try:
+            entity_info = await client.rest_get_async(f"/entity/{self.entity_id}")
+
+            # If this is a DockerRepository, fetch docker image tag & digest, and add it to the entity_info dict
+            if (
+                entity_info.get("concreteType")
+                == "org.sagebionetworks.repo.model.docker.DockerRepository"
+            ):
+                docker_tag_response = await client.rest_get_async(
+                    f"/entity/{self.entity_id}/dockerTag"
+                )
```

Comment:
Ideally all these calls can be put into a file (Or create a new one) here: https://github.com/Sage-Bionetworks/synapsePythonClient/tree/develop/synapseclient/api

Reply:
Sounds good! Initially was going to fold it into `entity_services.py`, but I think this would be incorrect since technically this service falls under handling the docker commits (/tags) and not the Docker Repository entities themselves ([API reference](https://rest-docs.synapse.org/rest/#org.sagebionetworks.repo.web.controller.DockerCommitController)). I've made a new file called `docker_commit_services.py` and stored it there.